services/src/kilda-pce/src/main/java/org/openkilda/pce/impl/AvailableNetwork.java

```diff
@@ -113,6 +118,43 @@ public void processDiversitySegments(List<PathSegment> segments) {
         }
     }
 
+    /**
+     * Adds diversity weights into {@link AvailableNetwork} based on passed path segments and configuration.
+     */
+    public void processDiversitySegmentsWithPop(List<PathSegment> segments) {
+        if (segments.size() <= 1) {
+            return;
+        }
+
+        Set<String> allocatedPopSet = new HashSet<>();
```

Comment:
nit: allocatedPopSet -> allocatedPops or transitPops

Reply:
fixed as we discussed offline